Build tensorflow-cc library with bazel and copy out the headers needed.

gartung opened the pull request.

cmsbuild commented:
A new Pull Request was created by  @gartung (Patrick Gartung) for branch IB/CMSSW_9_3_X/gcc630.

@cmsbuild, @smuzaffar, @mrodozov can you please review it and eventually sign? Thanks.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.

davidlange6 commented:
you should instead incorporate most of  this into tensorflow-sources.spec

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
https://cmssdt.cern.ch/jenkins/job/ib-any-integration/22496/console

Dr15Jones commented:
please abort test
Patrick is working on the code change recommended by David Lange.

cmsbuild commented:
Jenkins tests are aborted.

gartung commented:
This is an attempt to implement what was done with cmake here
https://github.com/FloopCZ/tensorflow_cc

gartung commented:
@davidlange6 The cmake config files want to link in libprotobuf.a when libtensorflow_cc.so is used. 
https://github.com/FloopCZ/tensorflow_cc/blob/master/tensorflow_cc/CMakeLists.txt#L69
What is the environment variable I need to set in the toolfile.xml to have that included?

cmsbuild commented:
Pull request #3362 was updated.

smuzaffar commented on tensorflow-cc-toolfile.spec:
```diff
@@ -0,0 +1,22 @@
+  <client>
+    <environment name="TENSORFLOW_CC_BASE" default="@TOOL_ROOT@"/>
+    <environment name="LIBDIR" default="$TENSORFLOW_CC_BASE/lib:$TENSORFLOW_C_BASE/lib"/>
+    <environment name="INCLUDE" default="$TENSORFLOW_CC_BASE/include:$TENSORFLOW_C_BASE/include"/>
```
tensorflow-c-toolfile is already setting 
 - $TENSORFLOW_C_BASE/lib
 - $TENSORFLOW_C_BASE/include
so you do not need to set them here. all you need to do is to add dependency on tensorflow-c i.e. cleanup the above two lines and add ```<use name="tensorflow-c"/>```

smuzaffar commented on tensorflow-cc-toolfile.spec:
```diff
@@ -0,0 +1,22 @@
+### RPM external tensorflow-cc-toolfile 1.0
+Requires: tensorflow-cc tensorflow-c-toolfile
```
you do not need to depend on tensorflow-c-toolfile

Dr15Jones commented:
@gartung I also found a simple shell script described here
https://stackoverflow.com/questions/33620794/how-to-build-and-use-google-tensorflow-c-api

gartung commented:
@davidlange6 @smuzaffar 
Both libtensorflow.so and libtensorflow_cc.so have 2 undefined references to Eigen:: symbols and many weak references to Eigen:: symbols.

>  nm -C libtensorflow*.so | grep "u Eigen::" 
>0000000000000000 u Eigen::NonBlockingThreadPoolTempl<tensorflow::thread::EigenEnvironment>::GetPerThread()::per_thread_
>0000000003d82630 u Eigen::internal::manage_caching_sizes(Eigen::Action, long*, long*, long*)::m_cacheSizes

>nm -C libtensorflow_cc.so | grep "W Eigen::"
>0000000002d37b40 W Eigen::QInt32 tensorflow::FloatToQuantized<Eigen::QInt32>(float, float, float)

The tensorflow build downloads many third party dependencies including Eigen and protobuf. I compiles the object files and links them into the library. This is probably OK when using libtensorflow.so with python. When using libtensorflow_cc.so in CMSSW it could potentially be loaded at the same time as the Eigen shared library causing symbol collisions.

Did you check for this when you built the tensorflow python package?

davidlange6 commented:
> On Aug 25, 2017, at 9:37 AM, Patrick Gartung <notifications@github.com> wrote:
> 
> @davidlange6 @smuzaffar
> Both libtensorflow.so and libtensorflow_cc.so have 2 undefined references to Eigen:: symbols and many weak references to Eigen:: symbols.
> 
> nm -C libtensorflow*.so | grep "u Eigen::"
> 0000000000000000 u Eigen::NonBlockingThreadPoolTempltensorflow::thread::EigenEnvironment::GetPerThread()::per_thread_
> 0000000003d82630 u Eigen::internal::manage_caching_sizes(Eigen::Action, long*, long*, long*)::m_cacheSizes
> 
> nm -C libtensorflow_cc.so | grep "W Eigen::"
> 0000000002d37b40 W Eigen::QInt32 tensorflow::FloatToQuantizedEigen::QInt32(float, float, float)
> 
> The tensorflow build downloads many third party dependencies including Eigen and protobuf. I compiles the object files and links them into the library. This is probably OK when using libtensorflow.so with python. When using libtensorflow_cc.so in CMSSW it could potentially be loaded at the same time as the Eigen shared library causing symbol collisions.
> 
> Did you check for this when you built the tensorflow python package?

Its a known issue with using tensorflow, yes.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

davidlt commented:
There is an upstream ticket and they are slowing working on this issue. TL;DR Linking to tensorflow C/C++ is unsafe with many reports from various distributions. This usually affects bigger projects with large dependencies tree.

cmsbuild commented:
Pull request #3362 was updated.

gartung commented:
@Dr15Jones Can you start the test.

Dr15Jones commented:
Please test

cmsbuild commented:
The tests are being triggered in jenkins.
https://cmssdt.cern.ch/jenkins/job/ib-any-integration/22501/console
https://cmssdt.cern.ch/jenkins/job/ib-any-integration/22503/console

gartung commented:
@Dr15Jones As far as I can tell Eigen is a header only package. See for instance
/cvmfs/cms.cern.ch/slc7_amd64_gcc630/external/eigen/3.3.0/
There is no library, just header files and config files for how to include them.

gartung commented:
@davidlange6 @Dr15Jones  this is the file to modify use our sources for eigen and protpbuf
https://github.com/tensorflow/tensorflow/blob/v1.1.0/tensorflow/workspace.bzl#L77
https://github.com/tensorflow/tensorflow/blob/v1.1.0/tensorflow/workspace.bzl#L227
There is no guarantee that our sources are compatible.

davidlange6 commented:
Hi Patrick
do you want to try to make a patch to see if our sources work?

> On Aug 25, 2017, at 1:07 PM, Patrick Gartung <notifications@github.com> wrote:
> 
> @davidlange6 @Dr15Jones this is the file to modify use our sources for eigen and protpbuf
> https://github.com/tensorflow/tensorflow/blob/v1.1.0/tensorflow/workspace.bzl#L77
> https://github.com/tensorflow/tensorflow/blob/v1.1.0/tensorflow/workspace.bzl#L227
> There is no guarantee that our sources are compatible.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

gartung commented:
I am working on a kludge right now which is to copy our header files into the eigen_archive directory that bazel created.

 

From: David Lange <notifications@github.com>
Reply-To: cms-sw/cmsdist <reply@reply.github.com>
Date: Friday, August 25, 2017 at 4:05 PM
To: cms-sw/cmsdist <cmsdist@noreply.github.com>
Cc: Patrick E Gartung <gartung@fnal.gov>, Mention <mention@noreply.github.com>
Subject: Re: [cms-sw/cmsdist] Build tensorflow-cc library with bazel and copy out the headers needed. (#3362)

 

Hi Patrick 
do you want to try to make a patch to see if our sources work? 

> On Aug 25, 2017, at 1:07 PM, Patrick Gartung <notifications@github.com> wrote: 
> 
> @davidlange6 @Dr15Jones this is the file to modify use our sources for eigen and protpbuf 
> https://github.com/tensorflow/tensorflow/blob/v1.1.0/tensorflow/workspace.bzl#L77 
> https://github.com/tensorflow/tensorflow/blob/v1.1.0/tensorflow/workspace.bzl#L227 
> There is no guarantee that our sources are compatible. 
> 
> — 
> You are receiving this because you were mentioned. 
> Reply to this email directly, view it on GitHub, or mute the thread. 
> 



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

Dr15Jones commented:
@davidlange6 instead of a patch, why not just fork our own tensorflow repository? That's how we now handle modifying 3rd party sources.

gartung commented:
@davidlange6 is there a way to have rpmbuild pull the sources used for the eigen and protobuf builds?

davidlange6 commented:
implementation detail. but lets see if something works as a first step
> On Aug 25, 2017, at 2:08 PM, Chris Jones <notifications@github.com> wrote:
> 
> @davidlange6 instead of a patch, why not just fork our own tensorflow repository? That's how we now handle modifying 3rd party sources.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

davidlange6 commented:
likely - but I don't know of an example that we have for it. presumably the tensorflow patch is to make the source derived from an envvar - a version for seeing what else goes wrong could hardwire the sources given the dependency and its version (from cmsrep)

> On Aug 25, 2017, at 2:14 PM, Patrick Gartung <notifications@github.com> wrote:
> 
> @davidlange6 is there a way to have rpmbuild pull the sources used for the eigen and protobuf builds?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
+1
Tested at: 3e6c7c09f27a6a0c78678703367a7e30a9be7665
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3362/22503/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3362/22503/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2653934
 - DQMHistoTests: Total failures: 210
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2653535
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

cmsbuild commented:
Pull request #3362 was updated.

gartung commented:
Just to note, when I changed workspace.bzl to use the CMS eigen source the build works. When I changed the protobuf source the build fails. I think this is because the CMS source is c++ only and the bazel build is expecting protobuf source for java as well.

gartung commented:
The version of eigen CMS uses is too new for tensorflow 1.1.0. I need to move to v1.2.1 or the just released 1.3.0:

tensorflow/core/lib/core/threadpool.cc:91:56: error: no matching function for call to 'Eigen::NonBlockingThreadPoolTempl<tensorflow::thread::EigenEnvironment>::NonBlockingThreadPoolTempl(int&, bool&, tensorflow::thread::EigenEnvironment)'
             EigenEnvironment(env, thread_options, name)) {}
                                                        ^
In file included from external/eigen_archive/unsupported/Eigen/CXX11/ThreadPool:58:0,
                 from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:72,
                 from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:4,
                 from tensorflow/core/lib/core/threadpool.cc:19:
external/eigen_archive/unsupported/Eigen/CXX11/src/ThreadPool/NonBlockingThreadPool.h:22:3: note: candidate: Eigen::NonBlockingThreadPoolTempl<Environment>::NonBlockingThreadPoolTempl(int, Environment) [with Environment = tensorflow::thread::EigenEnvironment]
   NonBlockingThreadPoolTempl(int num_threads, Environment env = Environment())
   ^~~~~~~~~~~~~~~~~~~~~~~~~~
external/eigen_archive/unsupported/Eigen/CXX11/src/ThreadPool/NonBlockingThreadPool.h:22:3: note:   candidate expects 2 arguments, 3 provided

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

gartung commented:
https://bitbucket.org/eigen/eigen/branches/compare/3.3.4%0Df3a22f35b044#pull-requests

gartung commented:
https://bitbucket.org/eigen/eigen/branches/compare/f3a22f35b044%0D3.3.4#diff

cmsbuild commented:
Pull request #3362 was updated.

davidlange6 commented:
since I understood the opposite in the meeting - this commit is in the eigen master branch...I would guess tensorflow just picked up the tip of the recipe at some moment in time (from late April). I don't see any fundamental reason against trying to move to it in CMSSW (other than if we have a second external package that requires a specific version of eigen). Lets start that in the devel branch..

> On Aug 29, 2017, at 5:38 PM, Patrick Gartung <notifications@github.com> wrote:
> 
> https://bitbucket.org/eigen/eigen/branches/compare/f3a22f35b044%0D3.3.4#diff
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

davidlange6 commented:
that said, I don't immediately see how fixes in the 3.3 branch get upstreamed. maybe you do?

> On Aug 29, 2017, at 7:36 PM, David Lange <David.Lange@cern.ch> wrote:
> 
> since I understood the opposite in the meeting - this commit is in the eigen master branch...I would guess tensorflow just picked up the tip of the recipe at some moment in time (from late April). I don't see any fundamental reason against trying to move to it in CMSSW (other than if we have a second external package that requires a specific version of eigen). Lets start that in the devel branch..
> 
>> On Aug 29, 2017, at 5:38 PM, Patrick Gartung <notifications@github.com> wrote:
>> 
>> https://bitbucket.org/eigen/eigen/branches/compare/f3a22f35b044%0D3.3.4#diff
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub, or mute the thread.
>> 
>

gartung commented:
Maybe the primary author merge changes from the 3.3 branch to default?

gartung commented:
@davidlange6 I tested this on my desktop and it builds. It needs to be tested with CMSSW.

cmsbuild commented:
Pull request #3362 was updated.

riga commented:
Hi.
In the context of cms-sw/cmssw#19893, are there any updates on this matter? Compared to the C API, the C++ version would do a great job providing (some of) the convenience to work with TF in CMSSW. However, if you find that right now the current setup process is not stable / compatible with cmsdist, we will of course continue to use the C version.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

gartung commented:
@davidlange6 @smuzaffar This builds properly once the eigen update is merged to 94x/gcc630.

slava77 commented:
please test

eigen #3435 was merged on Sept 20.
Does this PR need anything else?

cmsbuild commented:
The tests are being triggered in jenkins.
https://cmssdt.cern.ch/jenkins/job/ib-any-integration/23195/console

gartung commented:
no

gartung commented:
It needs the version of eigen on the 630next branch. This is a non tagged version from the default eigen branch.

gartung commented:
I see that was merged in the pull request you mentioned.

slava77 commented:
OK, all good then. 

it looks like jenkins ran out of workers for PR testing: this PR tests haven't started yet (and there is only one PR test in progress)

slava77 commented:
2 hours later, tests haven't started.
@smuzaffar @mrodozov @davidlange6 
is there some service interruption that prevents using more nodes for tests?

davidlange6 commented:
not answering the question - but it is the day that all the IBs get redone from scratch....

> On Sep 24, 2017, at 2:07 PM, Slava Krutelyov <notifications@github.com> wrote:
> 
> 2 hours later, tests haven't started.
> @smuzaffar @mrodozov @davidlange6
> is there some service interruption that prevents using more nodes for tests?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

slava77 commented:
On 9/24/17 10:09 AM, David Lange wrote:
> not answering the question - but it is the day that all the IBs get
> redone from scratch....

OK, this should explain.
I should just do something else on Sundays

cmsbuild commented:
-1

Tested at: 6e1eedac5b14f8a45e765adad52592b27859a19e

* **Build**:

I found an error when building: 
 
 <pre>PYTHON_LIB_PATH=/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/slc6_amd64_gcc630/external/py2-wheel/0.30.0a0-fmblme/lib/python2.7/site-packages \
TF_NEED_CUDA=0 \
TF_NEED_OPENCL=0 \
TMPDIR=/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/tmp \
external/bazel_tools/tools/cpp/link_dynamic_library.sh no ignored ignored ignored /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/slc6_amd64_gcc630/external/gcc/6.3.0/bin/gcc -shared -o bazel-out/local-opt/bin/tensorflow/libtensorflow.so -z defs -s -Wl,--version-script tensorflow/c/version_script.lds -pthread '-fuse-ld=gold' -Wl,-no-as-needed -Wl,-z,relro,-z,now -B/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/testBuildDir/slc6_amd64_gcc630/external/gcc/6.3.0/bin -B/usr/bin '-Wl,--build-id=md5' '-Wl,--hash-style=gnu' -pass-exit-codes -Wl,--gc-sections -Wl,@bazel-out/local-opt/bin/tensorflow/libtensorflow.so-2.params): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
bazel-out/local-opt/bin/tensorflow/core/kernels/_objs/attention_ops/tensorflow/core/kernels/attention_ops.pic.o:attention_ops.cc:function void Eigen::(anonymous namespace)::GlimpseExtractionOp<long>::eval<Eigen::TensorLayoutSwapOp<Eigen::TensorMap<Eigen::Tensor<float const, 4, 1, long>, 16, Eigen::MakePointer> const>, Eigen::TensorMap<Eigen::Tensor<float, 4, 0, long>, 0, Eigen::MakePointer>, Eigen::ThreadPoolDevice>(Eigen::TensorLayoutSwapOp<Eigen::TensorMap<Eigen::Tensor<float const, 4, 1, long>, 16, Eigen::MakePointer> const> const&, Eigen::TensorMap<Eigen::Tensor<float, 4, 0, long>, 0, Eigen::MakePointer>&, Eigen::ThreadPoolDevice const&) const: error: undefined reference to 'clock_gettime'
bazel-out/local-opt/bin/tensorflow/core/kernels/_objs/attention_ops/tensorflow/core/kernels/attention_ops.pic.o:attention_ops.cc:function void Eigen::(anonymous namespace)::GlimpseExtractionOp<long>::eval<Eigen::TensorLayoutSwapOp<Eigen::TensorMap<Eigen::Tensor<float const, 4, 1, long>, 16, Eigen::MakePointer> const>, Eigen::TensorMap<Eigen::Tensor<float, 4, 0, long>, 0, Eigen::MakePointer>, Eigen::ThreadPoolDevice>(Eigen::TensorLayoutSwapOp<Eigen::TensorMap<Eigen::Tensor<float const, 4, 1, long>, 16, Eigen::MakePointer> const> const&, Eigen::TensorMap<Eigen::Tensor<float, 4, 0, long>, 0, Eigen::MakePointer>&, Eigen::ThreadPoolDevice const&) const: error: undefined reference to 'clock_gettime'
collect2: error: ld returned 1 exit status
____Building complete.
Target //tensorflow/tools/lib_package:libtensorflow failed to build
____Elapsed time: 10.407s, Critical Path: 6.73s
</pre>

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3362/23195/summary.html

gartung commented:
I never tested this on SL6. That might be the source of this error:

error: undefined reference to 'clock_gettime'

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

davidlt commented:
Please fix your last two commits. The commit messages do not match the content.

We don't enable anything above SSE3 in CMSSW.  We still have some nodes which do not support SSE4.X (which is surprising as it's 2017 already). Unless you know this package will only be used on newer HW. You even enabled AVX, which landed in CPUs in ~2011. Double check the requirements and flags.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

gartung commented:
SSE4.1 and SSE4.2 have been available on AMD and Intel cpus since bdver1 and nehalem.

gartung commented:
The Fermilab people have run into this issue when using -march=core2

https://github.com/tensorflow/tensorflow/issues/12011

davidlt commented:
Correct, SSE4 stuff is old, but so are some SKUs in WLCG. It depends on particular world region. E.g., for optimized zlib we select functions based on hardware capability (HWCAP) to avoid this issue.

If you plan to use Ternsorflow on CPUs then you would probably highly benefit from SSE, AVX, AVX2, etc. IIRC, all CERN OpenStack [new] machines guarantee that you get AVX. If Tensorflow is good chunk of workflow then you might consider architecture specific builds and then just pick one based on HWCAP.

Solaris + SPARC has something like that built-in ( https://blogs.oracle.com/rie/shared-object-filters ). Used ISA extensions are recorded in ELF section and later dynamic loader could pick optimized one based on HWCAP.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

gartung commented:
@Dr15Jones @davidlange6 @smuzaffar I have fixed the SLC6 build and verified that tensorflow-sources build reliably. Can someone please issue the "please build" command.

slava77 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
https://cmssdt.cern.ch/jenkins/job/ib-any-integration/23239/console

cmsbuild commented:
-1

Tested at: 96bf4c3bf0c1b5a3e6deb7a5bc4b308d3f2e8441

* **Build**:

I found an error when building: 
 
 <pre>File "/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/PKGTOOLS/scheduler.py", line 199, in doSerial
result = commandSpec[0](*commandSpec[1:])
File "PKGTOOLS/cmsBuild", line 2816, in installPackage
File "PKGTOOLS/cmsBuild", line 2629, in installRpm
RpmInstallFailed: Failed to install package tensorflow-cc. Reason:
error: Failed dependencies:
	/usr/bin/python2.4 is needed by external+tensorflow-cc+1.3.0-1-1.x86_64

* The action "build-cms+cmssw-tool-conf+39.0-cms" was not completed successfully because The following dependencies could not complete:
install-external+tensorflow-cc-toolfile+1.0
</pre>

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3362/23239/summary.html

gartung commented:
OK, why did rpm decide that a no-source rpm has a dependence on /usr/bin/python2.4????????

* The action "install-external+tensorflow-cc+1.3.0" was not completed successfully because Traceback (most recent call last):
File "/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/PKGTOOLS/scheduler.py", line 199, in doSerial
result = commandSpec[0](*commandSpec[1:])
File "PKGTOOLS/cmsBuild", line 2816, in installPackage
File "PKGTOOLS/cmsBuild", line 2629, in installRpm
RpmInstallFailed: Failed to install package tensorflow-cc. Reason:
error: Failed dependencies:
	/usr/bin/python2.4 is needed by external+tensorflow-cc+1.3.0-1-1.x86_64

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

cmsbuild commented:
Pull request #3362 was updated.

gartung commented:
@slava77 can you start another test please

slava77 commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.
https://cmssdt.cern.ch/jenkins/job/ib-any-integration/23241/console

cmsbuild commented:
+1
Tested at: 3c7f62bd8effd8538290dbb0614227d509dfe76a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3362/23241/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3362/23241/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2698383
 - DQMHistoTests: Total failures: 232
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2697962
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 15 edm output root files, 26 DQM output files

gartung commented:
@davidlange6 @smuzaffar This needs to be added to a nightly build to resolve this problem
https://github.com/cms-sw/cmssw/pull/19893#issuecomment-332535700

slava77 commented:
@davidlange6
please clarify when can this be merged.
We are waiting for this to be able to proceed with  https://github.com/cms-sw/cmssw/pull/19893

slava77 commented:
ping

smuzaffar commented:
+externals

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next IB/CMSSW_9_4_X/gcc630 IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)
